File: Backend/Backend/Models/LLMLog.cs
using System.ComponentModel.DataAnnotations;
namespace Backend.Models;
public class LLMLog
{
    [Key]
    public long LogID { get; set; }
    public DateTime FechaHora { get; set; }
    public string TipoSolicitud { get; set; } = null!;
    public string Entrada { get; set; } = null!;
    public string Salida { get; set; } = null!;
    public decimal? CostoToken { get; set; }
    public int? DuracionMs { get; set; }
}

File: Flutter/lib/models/llm_log.dart
class LLMLog {
  final int logID;
  final DateTime fechaHora;
  final String tipoSolicitud;
  final String entrada;
  final String salida;
  final double? costoToken;
  final int? duracionMs;

  LLMLog({
    required this.logID,
    required this.fechaHora,
    required this.tipoSolicitud,
    required this.entrada,
    required this.salida,
    this.costoToken,
    this.duracionMs,
  });

  factory LLMLog.fromJson(Map<String, dynamic> json) => LLMLog(
        logID: json['logID'],
        fechaHora: DateTime.parse(json['fechaHora']),
        tipoSolicitud: json['tipoSolicitud'],
        entrada: json['entrada'],
        salida: json['salida'],
        costoToken:
            json['costoToken'] != null ? (json['costoToken'] as num).toDouble() : null,
        duracionMs: json['duracionMs'],
      );
}
